src/test/java/org/minions/devfund/lourdes/battleship/OceanTest.java

```diff
@@ -0,0 +1,92 @@
+package org.minions.devfund.lourdes.battleship;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Implemented tests for {@link Ocean}.
+ */
+public class OceanTest {
+    private Ocean ocean;
+
+    /**
+     * Method that setup initial values.
+     */
+    @Before
+    public void setup() {
+        ocean = new Ocean();
+    }
+
+    /**
+     * Verify the Ocean constructor.
+     */
+    @Test
+    public void testOceanInitialization() {
+        final String emptySea = "empty";
+        Ship[][] shipMatrix = ocean.getShipArray();
+        for (Ship[] row : shipMatrix) {
+            for (Ship cellValue : row) {
+                assertEquals(emptySea, cellValue.getShipType());
+            }
+        }
+    }
+
+    /**
+     * Verify if a given position is occupied.
+     */
+    @Test
+    public void testIsOccupied() {
+        final int row = 3;
+        final int column = 5;
+        final boolean horizontal = true;
+        Ship submarine = new Submarine();
+        submarine.placeShipAt(row, column, horizontal, ocean);
+        for (int i = column; i < submarine.length; i++) {
+            assertTrue(ocean.isOccupied(row, i));
+        }
+    }
+
+    /**
+     * Verify if a place is not occupied.
+     */
+    @Test
+    public void testIsNotOccupied() {
+        Ocean ocean = new Ocean();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "ocean" which hides the field declared at line 13. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)